llvm/test/MC/AArch64/SME2p2/fmop4as-fp16-fp32-widening-diagnostics.s

```diff
@@ -0,0 +1,243 @@
+// RUN: not llvm-mc -triple=aarch64 -mattr=+sme2p2 < %s 2>&1 | FileCheck %s
+
+// FMOP4A
+
+// Single vectors
+
+fmop4a za0.d, z0.h, z16.h
+// CHECK: [[@LINE-1]]:{{[0-9]+}}: error: invalid matrix operand
+
+fmop4a za4.s, z0.h, z16.h
+// CHECK: [[@LINE-1]]:{{[0-9]+}}: error: invalid operand for instruction
+
+fmop4a za0.s, z0.d, z16.h
+// CHECK: [[@LINE-1]]:{{[0-9]+}}: error: Invalid restricted vector register
+
+fmop4a za0.s, z15.h, z16.h
+// CHECK: [[@LINE-1]]:{{[0-9]+}}: error: Invalid restricted vector register
```

Comment:
It might be best to check for the whole error string (e.g, `expected even register in z0.h..z14.h`) when testing the restrictions we want from `ZPR16Mul2_Lo`, I'm not sure it appears in this file.

Reply:
If my memory serves me well, I deliberately omitted the part that mentions the specific register range as that depends on the compiler making a guess which would be the correct instruction and that guess depends on which instructions are implemented at the moment. Adding a new instruction can change the guess and thus the error message in effect breaking the test.

Reply:
I see, so it would arbitrarily guess which `fmop4a` we want meant after it fails to match any of them perfectly? I think I remember the same issue effecting the error messages for the `Za` registers.